akka-http-core/src/test/scala/akka/http/impl/engine/client/HostConnectionPoolSpec.scala

```diff
@@ -200,7 +200,7 @@ class HostConnectionPoolSpec extends AkkaSpecWithMaterializer(
 
           conn1.pushResponse(HttpResponse(entity = HttpEntity.Default(ContentTypes.`application/octet-stream`, 100, Source.empty)))
           val res = expectResponse()
-          res.entity.contentLengthOption.get shouldEqual 100
+          res.entity.contentLengthOption.get shouldEqual 0
```

Comment:
Shouldn't that option be None in these two HEAD scenarios rather than exist and have 0 length?

Reply:
I thought the same, but the way it is implemented, it points to the entity so there is indeed a length defined, at least for `Strict` and `Default` entities (unless I am mistaken, here it is `EmptyEntity` which is a special case of `Strict`). The difference in output is performed in the renderer.

Maybe it could be improved by also making changes at the model level, but I feel like this PR is already "large" for such a small change on the functional surface, so I am quite reluctant about it. What is your opinion?

Reply:
Yeah, let's not touch that.